web/src/components/PluginFlow/PluginFlow.tsx

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react'
+import { Modal, Form, Input, Alert } from 'antd'
+import { Cell } from '@antv/x6'
+import { useIntl } from 'umi'
+
+import FlowGraph from './components/FlowGraph'
+import Toolbar from './components/Toolbar'
+import { DEFAULT_CONDITION_PROPS, DEFAULT_PLUGIN_PROPS, DEFAULT_STENCIL_WIDTH, DEFAULT_TOOLBAR_HEIGHT, FlowGraphEvent } from './constants'
+import styles from './style.less'
+import ConfigPanel from './components/ConfigPanel'
+import PluginDetail from '../Plugin/PluginDetail'
+import { fetchList } from '../Plugin/service'
+
+type Props = {
+  chart: {
+    cells: Cell.Properties[];
+  };
+  readonly?: boolean;
+}
+
+type PluginProps = {
+  id: string;
+  name: string;
+  visible: boolean;
+  data: any;
+}
+
+type ConditionProps = {
+  id: string;
+  visible: boolean;
+  data: string;
+}
+
+const PluginFlow: React.FC<Props> = ({ chart, readonly = false }) => {
+  const { formatMessage } = useIntl()
+
+  // NOTE: To prevent from graph is not initialized
+  const [isReady, setIsReady] = useState(false)
+  const [plugins, setPlugins] = useState<PluginComponent.Meta[]>([])
+
+  const [pluginProps, setPluginProps] = useState<PluginProps>(DEFAULT_PLUGIN_PROPS)
+  const [conditionProps, setConditionProps] = useState<ConditionProps>(DEFAULT_CONDITION_PROPS)
+
+  const getContainerSize = () => {
```

Comment:
![image](https://user-images.githubusercontent.com/2106987/117104744-4301b700-adaf-11eb-869c-8e764b5af817.png)

Hi @liuxiran, if we disable the right scrollbar, due to we have many plugins on the left, some of them cannot show completely.

Reply:
1. When I try to collapse it and reopen it, I cannot reproduce that the card gets smaller 🤔

Reply:
> ![image](https://user-images.githubusercontent.com/2106987/117104744-4301b700-adaf-11eb-869c-8e764b5af817.png)
> 
> Hi @liuxiran, if we disable the right scrollbar, due to we have many plugins on the left, some of them cannot show completely.

o I mean is it possible to narrow the hight of the display container so that we can not enable page right scroll bar, for both left and right container have their own scroll bar ^_^

Reply:
> 1. When I try to collapse it and reopen it, I cannot reproduce that the card gets smaller 🤔

my browser is chrome, and version is `90.0.4430.93（正式版本） (x86_64)`

and the the appearance of bug just like this:

https://user-images.githubusercontent.com/2561857/117374665-22954200-af00-11eb-99cd-17a195e1ac68.mp4

Reply:
got it, this issue would be addressed in another PR, let me add it to roadmap